lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -251,6 +251,63 @@ def add_merchant_description(post, options)
         merchant[:locality] = options[:merchant_descriptor_locality] if options[:merchant_descriptor_locality]
       end
 
+      def add_stored_credentials(post, payment, options)
+        return unless stored_credential = options[:stored_credential]
+
+        options = stored_credential_options(stored_credential, options.fetch(:reason_code, ''))
+        post[:processingInformation][:commerceIndicator] = options.fetch(:transaction_type, 'internet')
+        stored_credential[:initial_transaction] ? initial_transaction(post, options) : subsequent_transaction(post, options)
+      end
+
+      def stored_credential_options(options, reason_code)
+        transaction_type = options[:reason_type]
+        transaction_type = 'install' if transaction_type == 'installment'
+        initiator = options[:initiator] if  options[:initiator]
+        initiator = 'customer' if initiator == 'cardholder'
+        stored_on_file = options[:reason_type] == 'recurring'
+        options.merge({
+          transaction_type: transaction_type,
+          initiator: initiator,
+          reason_code: reason_code,
+          stored_on_file: stored_on_file
+        })
+      end
+
+      def add_processing_information(initiator, merchant_initiated_transaction_hash = {})
+        {
+          authorizationOptions: {
+            initiator: {
+              type: initiator,
+              merchantInitiatedTransaction: merchant_initiated_transaction_hash,
+              storedCredentialUsed: true
+            }
+          }
+        }.compact
+      end
+
+      def initial_transaction(post, options)
+        processing_information = add_processing_information(options[:initiator], {
+          reason: options[:reason_code]
+        })
+
+        post[:processingInformation].merge!(processing_information)
+      end
+
+      def subsequent_transaction(post, options)
+        network_transaction_id = options[:network_transaction_id] || options.dig(:stored_credential, :network_transaction_id) || ''
```

Comment:
Do we know it is proper to send this field with an empty string, rather than to omit the field, in the absence of an NTID?

Reply:
not really, I send empty string instead send a null value